e2e/test/scenarios/filters-reproductions/filters-reproductions.cy.spec.js

```diff
@@ -1513,3 +1513,28 @@ describe("Issue 48851", () => {
     popover().button("Add filter").should("be.visible");
   });
 });
+
+describe("issue 49321", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsNormalUser();
+  });
+
+  it("should not require multiple clicks to apply a filter (metabase#49321)", () => {
+    openProductsTable({ mode: "notebook" });
+    filter({ mode: "notebook" });
+    popover().within(() => {
+      cy.findByText("Title").click();
+      cy.findByText("Is").click();
+    });
+    popover().last().findByText("Contains").click();
+
+    popover().invoke("width").should("eq", 380);
+
+    popover()
+      .findByPlaceholderText("Enter some text")
+      .type("aaaaaaaaaa, bbbbbbbbbbb,");
+
+    popover().invoke("width").should("eq", 380);
```

Comment:
> It's also hard to compare the width of one element at one time with it's width at a different time in the test.

```suggestion
    popover().invoke("width").as("initialWidth");

    popover()
      .findByPlaceholderText("Enter some text")
      .type("aaaaaaaaaa, bbbbbbbbbbb,");

    cy.get("@initialWidth").then(initialWidth => {
      popover().invoke("width").should("eq", initialWidth);
    });
```

or

```suggestion
    popover().then($popover => {
      const { width } = $popover[0].getBoundingClientRect();
      cy.wrap(width).as("initialWidth");
    });

    popover()
      .findByPlaceholderText("Enter some text")
      .type("aaaaaaaaaa, bbbbbbbbbbb,");

    cy.get("@initialWidth").then(initialWidth => {
      popover().should($popover => {
        const { width } = $popover[0].getBoundingClientRect();
        expect(width).to.eq(initialWidth);
      });
    });
```

Reply:
Thanks! I tried the first option, but that does not reproduce the actual issue (I think because `invoke` gets retried until the correct width is returned).

The second option works well though!

Reply:
> I tried the first option, but that does not reproduce the actual issue

Hmm, maybe the first `then` makes the test flaky. Second option would suffer from the same problem.
I started a stress test to be sure: https://github.com/metabase/metabase/actions/runs/11680143931

Reply:
Seems stable!

Reply:
All green :tada: